### PR TITLE
Adding Hashtables

### LINKED ISSUE #127: [Feature Request] <Adding Hash tables>
### Is your feature request related to a problem? Please describe.
A clear and concise description of what the problem is. Example: "I'm frustrated when..."
I want to create a new section which is hash tables, as it is a very important topic in Data structures. 

### Describe the solution you'd like
It would be about explanation of hash-tables, hash maps etc.

### Describe alternatives you've considered
Instead of adding a dedicated section for hash tables, I considered including it as part of a broader "Advanced Data Structures" module that would cover multiple complex structures like trees, graphs, and hash tables together. However, hash tables are important enough to warrant their own focused section. 

### Additional context
image

---

**Would you like to work on this feature?**  
- [.] Yes  I would love to work on this feature, Please assign this to me
- [ ] No

### PR DESCRIPTION
**Pull request by Mahateaa**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes https://github.com/ajay-dhangar/algo/issues/127

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/hash/Hashtables.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+---
+id: Hash-tables-Dsa
```
Comment (ajay-dhangar) [inline docs/hash/Hashtables.md]: replace `id: Hash-tables-Dsa` with `id: hash-tables-dsa` for good practice

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please update your code according to the provided suggestions. Additionally, rename the following files: 

- `docs/hash/Hashtables.md` to `docs/hash/hash-tables.md`
- `docs/hash/Hashcategory.json` to `docs/hash/_category_.json`

Ensure that your changes align with our documentation structure and best practices. For detailed guidance, refer to [Docusaurus Documentation](https://docusaurus.io/docs/).

File: docs/hash/Hashtables.md (hunk 1 above)
Comment (Mahateaa) [reply docs/hash/Hashtables.md]: only changing the id and renaming the files is enough or should I check on the code too?

Comment (ajay-dhangar) [reply docs/hash/Hashtables.md]: You need to sync our latest code. I've resolved the error, so first, sync with `git pull origin main`, push the code to your branch, and then resolve any remaining errors.

Comment (Mahateaa) [reply docs/hash/Hashtables.md]: Done can you check now?

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
recheck your `docs/hash/hash-tables.md` file code because it's syntax error

**Comment by Mahateaa** [comment]
can you please check again, have updated the code

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**